packages/stylelint-config/index.js

```diff
@@ -43,6 +39,21 @@ module.exports = {
       },
     ],
 
+    // Custom regex for ReadMe id patterns.
+    'selector-id-pattern': [
+      '^(([a-z][a-z0-9]*(-[a-z0-9]+)*)|([A-Z][a-z0-9]*)+)$',
+      {
+        message: 'Expected id selector to be kebab-case or TitleCase',
+      },
+    ],
+
+    // ReadMe still references elements by #id. Allow at most one per selector.
+    'selector-max-id': 1,
+
+    // ReadMe relies on "element[attr='value']" selectors in too many places.
```

Comment:
Interesting. We use these kinds of custom attribute selectors a lot for things like our `<Flex>` component. I personally find them useful/cool! Why does stylelint not like ‘em?

Reply:
Right, I was surprised to see this rule tightened up too. But after thinking about it, I do see how it is good practice to avoid themm.

The following selectors in practical use are mostly redundant:
```css
input[type=password] {
  // Style my password input field
}

[type=password] {
  // Style my password field
}
```

I think type vs attr selectors also have different levels of specificity. So mixing and matching can create confusing descending specificity issues.

The majority of the time, I think you can mostly get away without combining both tag + attr selectors. And if you need to target an element (say inside a component), it kinda forces you to do the proper legwork and assign the element a proper BEM or module class name which you then target instead of the tag + attr combination.